backend/picamera_backend.py

```diff
@@ -30,18 +30,21 @@ class PiCameraBackend(Backend):
         This backend will only work when the `picamera` library is available,
         i.e. when running on a Raspberry Pi. """
 
-    def __init__(self, width: int, height: int, publish_recording_status_func: PublishFunc, exception_handler: PublishFunc):
-        super().__init__(width, height, publish_recording_status_func, exception_handler)
+    def __init__(self, width: int, height: int, publish_recording_status_func: PublishFunc, publish_camera_is_online_func: PublishFunc, exception_handler: PublishFunc):
+        super().__init__(width, height, publish_recording_status_func, publish_camera_is_online_func, exception_handler)
 
         if not ON_PI:
             raise RuntimeError("`picamera` library unavailable - please run on Pi or install library")
-        
+
         self.pi_camera = PiCamera(resolution=(self.width, self.height))
 
         self.prev_overlays: Dict[PiCameraOverlayLayer, self.pi_camera.PiOverlayRenderer] = {}
 
+    def _is_camera_on(self):
+        return ON_PI
```

Comment:
`ON_PI` only specifies whether the code is running on a raspberry pi or not. `return self.pi_camera.previewing` should do the trick!

Reply:
Done
0cf8f0c